src/feditest/nodedrivers/mastodon/__init__.py

```diff
@@ -45,15 +46,15 @@ def _token_validate(candidate: str) -> str | None:
     return candidate if len(candidate)>10 else None
 
 
+@dataclass
 class UserRecord:
     """
     Collects what we know of a user at a NodeWithMastodonAPI
     """
-    def __init__(self, userid: str, email: str, passwd: str):
-        self._userid = userid
-        self._email = email
-        self._passwd = passwd
-        self._mastodon_user_client = None
+    _userid : str
```

Comment:
+1 on the dataclass change, but we should rename these fields without the leading underscore. It it's important, you can make the fields init-only using `@dataclass(frozen=True)` decorator option.